utils/promise.js

```diff
@@ -1,21 +1,26 @@
-export const makeCancelable = (promise, options) => {
-  let canceled = false
-
-  const wrappedPromise = new Promise((resolve, reject) =>
+export function CancelablePromise(promise, options) {
```

Comment:
Why it couldn't be canceled? It will work for any promise not only for fetch. The abstraction is correct IMHO. You can optionally pass a specific cancel function. 🙂